File: result/repair/transcoder/java2python/python_repair/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP_1.py
def f_gold ( arr , n ) :
    i=0 
    while i<(n-2)/2+1 :
        if arr [ 2 * i + 1 ] > arr [ i ] :
            return False
        if 2 * i + 2 < n and arr [ 2 * i + 2 ] > arr [ i ] :
            return False
        i += 1
    return True

File: result/translate/starchat/python2java/java/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP_1.java
public static boolean f_gold ( int arr [ ], int n ) {
        for ( int i = 0 ; i < ( n - 2 ) / 2 + 1 ; i++ ) {
            if ( arr [ 2 * i + 1 ] > arr [ i ] ) return false;
            if ( 2 * i + 2 < n && arr [ 2 * i + 2 ] > arr [ i ] ) return false;
        }
        return true;
    }